src/syncerd/types.rs

```diff
@@ -348,26 +423,63 @@ pub struct TransactionConfirmations {
     pub tx: Vec<Vec<u8>>,
 }
 
-#[derive(Clone, Debug, Display, StrictEncode, StrictDecode, Eq, PartialEq, Hash)]
-#[display(Debug)]
+impl fmt::Display for TransactionConfirmations {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(
+            f,
+            "TransactionConfirmations(id: {}, block: {}, confirmations: {:?})",
+            self.id,
+            hex::encode(&self.block),
+            self.confirmations,
+        )
+    }
+}
+
+#[derive(Clone, Debug, StrictEncode, StrictDecode, Eq, PartialEq, Hash)]
 pub struct TransactionBroadcasted {
     pub id: TaskId,
     pub tx: Vec<u8>,
     pub error: Option<String>,
 }
 
-#[derive(Clone, Debug, Display, StrictEncode, StrictDecode, Eq, PartialEq, Hash)]
-#[display(Debug)]
+impl fmt::Display for TransactionBroadcasted {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        let bitcoin_tx_id =
+            bitcoin::Transaction::consensus_decode(std::io::Cursor::new(self.tx.clone()))
+                .map(|tx| tx.txid().to_string())
+                .unwrap_or_default();
+        write!(
+            f,
+            "TransactionBroadcasted(id: {}, tx_id: {}, error: {:?})",
+            self.id, bitcoin_tx_id, self.error,
+        )
+    }
+}
+
+#[derive(Clone, Debug, StrictEncode, StrictDecode, Eq, PartialEq, Hash)]
 pub struct SweepSuccess {
     pub id: TaskId,
-    pub txids: Vec<Vec<u8>>,
+    pub txids: Vec<Txid>,
+}
+
+impl fmt::Display for SweepSuccess {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(
+            f,
+            "SweepSuccess(id: {}, txids: {:?})",
+            self.id,
+            self.txids
+                .iter()
+                .map(|t| t.to_string())
```

Comment:
Maybe `.join(", ")` and default formatter

Reply:
bc1a263263707d47351cf0d60551955262e2d670